client/app/states/catalogs/details/details.state.js

```diff
@@ -81,9 +82,11 @@ function Controller($stateParams, CollectionsApi, EventNotifications, DialogFiel
         vm.dialogUrl = `service_catalogs/${vm.serviceTemplate.service_template_catalog_id}/service_templates`;
         if (dialogs.subcount > 0) {
           DialogFieldRefresh.setupDialogData(vm.dialogs, allDialogFields, autoRefreshableDialogFields);
-                   
           angular.forEach(allDialogFields, function (dialogField) {
-            dialogField.refreshSingleDialogField = function () {
+            dialogField.refreshSingleDialogField = function (button) {
+              if (button && !dialogField.load_values_on_init) {
+                dialogField.load_values_on_init = true;
+              }
```

Comment:
I think this is fine but I want to clarify, so for something that has `dialogField.load_values_on_init` set to false initially, if we click on the refresh button, this will cause the field to refresh and then set `load_values_on_init` to true. I don't think this would cause any weird side effects because we've already gone past the "init" stage, right?

Reply:
This shouldn't cause any weird side effects because it would have been initialized once by virtue of clicking on refresh.